stdlib/http/src/main/ballerina/http/resiliency/http_circuit_breaker.bal

```diff
@@ -711,32 +711,152 @@ documentation {
+        totalCount  =  totalCount + bucket.totalCount;
+    }
+    return totalCount;
+}
+
+documentation {
+    Calculate the current bucket Id.
+
+    P{{circuitHealth}}  Circuit Breaker health status
+    P{{circuitBreakerInferredConfig}}   Configurations derived from `CircuitBreakerConfig`
+    R{{}} Current bucket id
+}
+function getCurrentBucketId(CircuitHealth circuitHealth, CircuitBreakerInferredConfig circuitBreakerInferredConfig)
+             returns int {
+    int elapsedTime = (time:currentTime().time - circuitHealth.startTime.time) % circuitBreakerInferredConfig.
+        rollingWindow.timeWindowMillis;
+    int currentBucketId = ((elapsedTime / circuitBreakerInferredConfig.rollingWindow.bucketSizeMillis) + 1)
+        % circuitBreakerInferredConfig.noOfBuckets;
+    return currentBucketId;
+}
+
+documentation {
+    Update rejected requests count.
+
+    P{{circuitHealth}}  Circuit Breaker health status
+    P{{circuitBreakerInferredConfig}}   Configurations derived from `CircuitBreakerConfig`
+}
+function updateRejectedRequestCount(CircuitHealth circuitHealth, CircuitBreakerInferredConfig circuitBreakerInferredConfig) {
+    int currentBucketId = getCurrentBucketId(circuitHealth, circuitBreakerInferredConfig);
+    updateLastUsedBucketId(currentBucketId, circuitHealth);
+    circuitHealth.totalBuckets[currentBucketId].rejectedCount++;
+}
+
 documentation {
     Reset the bucket values to default ones.
 
     P{{circuitHealth}}  - Circuit Breaker health status.
     P{{bucketId}}  - Id of the bucket should reset.
 }
 function resetBucketStats(CircuitHealth circuitHealth, int bucketId) {
-    circuitHealth.totalBuckets[bucketId].successCount = 0;
+    circuitHealth.totalBuckets[bucketId].totalCount = 0;
     circuitHealth.totalBuckets[bucketId].failureCount = 0;
+    circuitHealth.totalBuckets[bucketId].rejectedCount = 0;
 }
 
 function getEffectiveErrorTime(CircuitHealth circuitHealth) returns time:Time {
     return (circuitHealth.lastErrorTime.time > circuitHealth.lastForcedOpenTime.time)
-                        ? circuitHealth.lastErrorTime : circuitHealth.lastForcedOpenTime;
-}
\ No newline at end of file
+    ? circuitHealth.lastErrorTime : circuitHealth.lastForcedOpenTime;
+}
+
+documentation {
+    Populate the `RollingWindow` statistics to handle circuit breaking within the `RollingWindow` time frame.
+
+    P{{circuitHealth}}  Circuit Breaker health status
+    P{{circuitBreakerInferredConfig}}   Configurations derived from `CircuitBreakerConfig`
+}
+function prepareRollingWindow(CircuitHealth circuitHealth, CircuitBreakerInferredConfig circuitBreakerInferredConfig) {
+
+    int currentTime = time:currentTime().time;
+    int idleTime = currentTime - circuitHealth.lastRequestTime.time;
+
+    // If the time duration between Two requests greater than timeWindowMillis values, reset the buckets to default.
+    if (idleTime > circuitBreakerInferredConfig.rollingWindow.timeWindowMillis) {
```

Comment:
Shall we add a private variable for circuitBreakerInferredConfig.rollingWindow and reuse it within the function